x-pack/plugins/security_solution/server/endpoint/services/artifacts/manifest_manager/manifest_manager.ts

```diff
@@ -49,36 +52,34 @@ interface BuildArtifactsForOsOptions {
   name: string;
 }
 
-const iterateArtifactsBuildResult = async (
+const iterateArtifactsBuildResult = (
   result: ArtifactsBuildResult,
-  callback: (artifact: InternalArtifactCompleteSchema, policyId?: string) => Promise<void>
+  callback: (artifact: InternalArtifactCompleteSchema, policyId?: string) => void
 ) => {
   for (const artifact of result.defaultArtifacts) {
-    await callback(artifact);
+    callback(artifact);
   }
 
   for (const policyId of Object.keys(result.policySpecificArtifacts)) {
     for (const artifact of result.policySpecificArtifacts[policyId]) {
-      await callback(artifact, policyId);
+      callback(artifact, policyId);
     }
   }
 };
 
 const iterateAllListItems = async <T>(
   pageSupplier: (page: number) => Promise<ListResult<T>>,
-  itemCallback: (item: T) => void
+  itemCallback: (items: T[]) => void
 ) => {
   let paging = true;
   let page = 1;
 
   while (paging) {
     const { items, total } = await pageSupplier(page);
 
-    for (const item of items) {
-      await itemCallback(item);
-    }
+    itemCallback(items);
 
-    paging = (page - 1) * 20 + items.length < total;
+    paging = (page - 1) * 1000 + items.length < total;
```

Comment:
This causes me confusion (it may just be me ). You are assuming that the page size that `pageSupplier()` will use is `1000`, but really, you can't be sure of that because one could alter the callbasck given to this function. That makes this a bit fragile because someone could (unknownly) change how `pageSupplier()` works and not realize that this is assuming a page size of `1000` here - thus possibly causing items to be dropped or duplicated.

Suggestion: change `pageSupplier()` to take in as arguments the `page` and the `perPage` arguments.

Reply:
Very good point! Thanks!

Reply:
https://github.com/elastic/kibana/pull/160387/commits/5c9ef42bc40abfde1ec0b496a2292b3809bba6ba